### PR TITLE
improve rule docs

### PR DESCRIPTION
**Pull request by rafaqz**
# Description

Documentation for rule constructors and keyword arguments, and cleanup of old docs and doc workflows.

The examples are still broken because some media files are missing that they need. Will fix examples in another PR.

## Changes :

- [ ] New formulation/rule
- [x] Documentation
- [ ] Bug fix for existing rules (non-breaking syntax)
- [ ] Major change to existing rules (breaking syntax, needs a breaking version bump)

### CODE REVIEW

**Comment by rafaqz** [comment]
@jamesmaino I'm keen to start doing reviews for all PRs here from now on. This one has a bit of technical stuff with the documentation CI, but it would be good to have you and virgile accross this stuff in future anyway, it's not really very complicated.

Besides new github workflows to make these docs actually build, and deleting old things, mostly this restructures the docs so they are easier to read and use to define rules, but it would be good to get a second opinion on if it achieves that.

**Comment by rafaqz** [comment]
@jamesmaino can you have a quick look and approve, just to get this review ball rolling. This as no consequence on models its just docs so its low pressure ;). 

I've got some more updates to come after this that will be easier once this is merged.

**Review by jamesmaino — APPROVED** [review]

**Pull request merged**
